Use the same Python interpreter that the passed-in Meson does.

--- configure.orig
+++ configure
@@ -1479,6 +1479,13 @@
 # Remove old dependency files to make sure that they get properly regenerated
 rm -f */config-devices.mak.d

+# If --meson is supplied but --python is not, extract the Python interpreter
+# from the Meson script
+if test "$explicit_python" = no && ! test -z "$meson" && test "$meson" != "git" && test "$meson" != "internal"
+then
+    python=$(sed -n '1s/^#!//p' $(command -v "$meson"))
+fi
+
 if test -z "$python"
 then
     error_exit "Python not found. Use --python=/path/to/python"
